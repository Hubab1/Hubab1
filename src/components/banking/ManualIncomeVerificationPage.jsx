import React from 'react';
import styled from '@emotion/styled';

import { BackButton } from 'components/common/BackLink';
import ActionButton from 'components/common/ActionButton/ActionButton';
import Capsule from 'components/common/Capsule/Capsule';
import { H1, H3 } from 'assets/styles';
import finance from 'assets/images/finance.png';

const SkinnyH1 = styled(H1)`
    width: 70%;
`;

const SpacedH3 = styled(H3)`
    margin-top: 15px;
    margin-bottom: 30px;
`;

export default function ManualIncomeVerificationPage (props) {
    return (
        <>
            <SkinnyH1>Income and Asset Verification</SkinnyH1>
            <SpacedH3>Add at least one income source or asset below.</SpacedH3>
            <Capsule
                prefix={<img alt="coin" src={finance}></img>}
                label="Income"
                buttonLabel="Add an Income Source"
                tip="TBD"
            />
            <ActionButton marginTop={40} marginBottom={20}>
                Continue
            </ActionButton>
            <BackButton onClick={props.goBack} />
        </>
    );
}
