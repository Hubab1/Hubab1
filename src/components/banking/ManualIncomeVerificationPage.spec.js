import React from 'react';
import { shallow } from 'enzyme';
import { ManualIncomeVerificationPage } from './ManualIncomeVerificationPage';

jest.mock("react", () => ({
    ...jest.requireActual("react"),
    useContext: () => ({
        bankingData: {
            income_sources: [
                {
                    id: 123,
                    income_or_asset_type: 105,
                    estimated_amount: 100000
                }
            ],
            asset_sources: [
                {
                    id: 123,
                    income_or_asset_type: 510,
                    estimated_amount: 40000
                }
            ]
        },

    })
  }));

let defaultProps;
beforeEach(() => {
    defaultProps = {
        goBack: jest.fn()
    }
})

it('matches snapshot with some financial sources data', () => {
    const wrapper = shallow(<ManualIncomeVerificationPage {...defaultProps}/>);
    expect(
        wrapper.getElement()
    ).toMatchSnapshot();
});
