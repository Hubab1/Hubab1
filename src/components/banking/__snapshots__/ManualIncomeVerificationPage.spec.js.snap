// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot with some financial sources data 1`] = `
<React.Fragment>
  <ForwardRef(render)>
    Income and Asset Verification
  </ForwardRef(render)>
  <ForwardRef(render)>
    Add at least one income source or asset below.
  </ForwardRef(render)>
  <Capsule
    buttonLabel="Add an Income Source"
    expansionPanel={
      <ExistingItemsExpansionPanel
        label="Income Source"
        labelQuantity={1}
      >
        <div>
          <div>
            Employment
          </div>
          <div
            className="css-dufvih"
          >
            $100,000
            /year
          </div>
          <ForwardRef
            style={
              Object {
                "textDecoration": "underline",
              }
            }
            to="/income-employment/manual/financial-sources/123/edit"
          >
            Edit
          </ForwardRef>
        </div>
      </ExistingItemsExpansionPanel>
    }
    label="Income"
    limitReached={false}
    prefix={
      <img
        alt="coin"
        src="finance.png"
      />
    }
    route="/income-employment/manual/add-income"
    tip="TBD"
  />
  <Capsule
    buttonLabel="Add an Asset"
    expansionPanel={
      <ExistingItemsExpansionPanel
        label="Asset"
        labelQuantity={1}
      >
        <div>
          <div>
            Investment
          </div>
          <div
            className="css-dufvih"
          >
            $40,000
          </div>
          <ForwardRef
            style={
              Object {
                "textDecoration": "underline",
              }
            }
            to="/income-employment/manual/financial-sources/123/edit"
          >
            Edit
          </ForwardRef>
        </div>
      </ExistingItemsExpansionPanel>
    }
    label="Assets"
    limitReached={false}
    prefix={
      <img
        alt="piggy bank"
        src="piggy-bank.png"
      />
    }
    route="/income-employment/manual/add-asset"
    tip="TBD"
  />
  <ActionButton
    color="primary"
    marginBottom={20}
    marginTop={40}
    successGreen={false}
    variant="contained"
  >
    Continue
  </ActionButton>
  <BackLink
    to="/income-employment"
  />
</React.Fragment>
`;
